src/payment/strategies/cardinal/cardinal-client.ts

```diff
@@ -22,24 +22,33 @@ export interface CardinalOrderData {
 }
 
 export default class CardinalClient {
+    private _provider: string = '';
+    private _testMode: boolean = false;
     private _sdk?: Promise<CardinalSDK>;
-    private _isConfigured: boolean = false;
+    private _configurationToken: string = '';
 
     constructor(
         private _scriptLoader: CardinalScriptLoader
     ) {}
 
-    load(provider: string, testMode?: boolean): Promise<void> {
+    load(provider: string, testMode = false): Promise<void> {
+        this._provider = provider;
+        this._testMode = testMode;
+
         if (!this._sdk) {
             this._sdk = this._scriptLoader.load(provider, testMode);
         }
 
         return this._sdk.then(() => {});
     }
 
-    configure(clientToken: string): Promise<void> {
-        if (this._isConfigured) {
-            return Promise.resolve();
+    async configure(clientToken: string): Promise<void> {
+        if (!!this._configurationToken) {
```

Comment:
wouldn't `this._configurationToken` always be null? I think you are missing  `this._configurationToken = clientToken;` somewhere.

Maybe something like this?
```js
if (this._configurationToken && this._configurationToken === clientToken) {
  return Promise.resolve();
}

this._configurationToken = clientToken;
this._sdk = this._scriptLoader.load(`${this._provider}.${Date.now()}`, 
```

Reply:
Yes but, I just wanted to make sure to load it when a reconfiguration took place 😬. Otherwise, we may end up with duplicated/wasted scripts the first time `configure` runs because it's supposed to be already loaded by the `prepare` method of` CardinalThreeDSecureFlow[v2]`.